src/resources/js/main.ts

```diff
@@ -75,27 +100,34 @@ function jsApiFunction() {
                         <td>${person.hair_color}</td>
                         <td>${person.eye_color}</td>
                         `;
-        table.append(tr);
+        table!.append(tr);
     }
 
-    function showLoader(nodeIdToAppend) {
+    function showLoader(nodeIdToAppend: string) {
         let node = document.getElementById(nodeIdToAppend);
         
         let loader = document.createElement('span');
         loader.id = nodeIdToAppend + '-loader';
         loader.className = 'loader';
-        node.append(loader);
+        node!.append(loader);
     }
 
-    function hideLoader(nodeId) {
+    function hideLoader(nodeId: string) {
         let loader = document.getElementById(nodeId + '-loader');
-        loader.remove();
+        loader!.remove();
     } 
 
+    type Person = {
```

Comment:
I'd say this should go at the top of the file if you can. I like to put all my const/type definitions at the top. I try and rely on hoisting as little as possible

Reply:
Agreed. I'm not sure how I ended up with it at the bottom, but not the best place for it.